src/js/disability-benefits/utils/helpers.js

```diff
@@ -59,40 +59,60 @@ export function getUserPhase(phase) {
   return phase - 3;
 }
 
+export function getSubmittedItemDate(item) {
+  if (item.receivedDate) {
+    return item.receivedDate;
+  } else if (item.documents && item.documents.length) {
+    return item.documents[item.documents.length - 1].uploadDate;
+  } else if (item.type === 'other_documents_list') {
+    return item.uploadDate;
+  }
+
+  return item.date;
+}
+
+function getPhaseNumber(phase) {
+  return parseInt(phase.replace('phase', ''), 10);
+}
+
+function isEventOrPrimaryPhase(event) {
+  if (event.type === 'phase_entered') {
+    return event.phase <= 3 || event.phase >= 7;
+  }
+
+  return !!getSubmittedItemDate(event);
+}
+
 export function groupTimelineActivity(events) {
   const phases = {};
-  const phaseEvents = events;
   let activity = [];
-  let lastPhaseNumber = 1;
-  let firstPhase = true;
 
-  phaseEvents.forEach(event => {
-    if (event.type.startsWith('phase')) {
-      const phaseNumber = parseInt(event.type.replace('phase', ''), 10);
-      const userPhaseNumber = getUserPhase(phaseNumber);
-      if (userPhaseNumber !== lastPhaseNumber || firstPhase) {
-        activity.push({
+  const phaseEvents = events
+    .map(event => {
+      if (event.type.startsWith('phase')) {
+        return {
           type: 'phase_entered',
+          phase: getPhaseNumber(event.type) + 1,
           date: event.date
-        });
-        phases[userPhaseNumber + 1] = (phases[userPhaseNumber + 1] || []).concat(activity);
-        activity = [];
-        lastPhaseNumber = userPhaseNumber;
-        firstPhase = false;
-      } else {
-        activity.push({
-          type: 'micro_phase',
-          phaseNumber: phaseNumber + 1,
-          date: event.date
-        });
+        };
       }
+
+      return event;
+    })
+    .filter(isEventOrPrimaryPhase);
+
+  phaseEvents.forEach(event => {
+    if (event.type.startsWith('phase')) {
+      activity.push(event);
+      phases[getUserPhase(event.phase)] = activity;
```

Comment:
Events that happen in phase 4-6 will be lost here when the phase3 even is parsed, as they all map to the same user phase (3). I think you want something like you had before
`(phases[getUserPhase(event.phase)] || []).concat(activity);`